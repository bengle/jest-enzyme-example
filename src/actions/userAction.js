import { get } from "../utils/http";
export const changeUsername = name => {
  return {
    type: "CHANGE_USERNAME",
    payload: name
  };
};
export const changeUserSex = isChecked => {
  return {
    type: "CHANGE_SEX",
    payload: isChecked
  };
};

export const getAllUsers = () => dispatch => {
  return get("http://www.mocky.io/v2/5c7658db3200000b24f46080").then(res => {
    if (res.data.code === 0) {
      dispatch({
        type: "GET_ALL_USERS",
        payload: res.data.data
      });
    }
  });
};
