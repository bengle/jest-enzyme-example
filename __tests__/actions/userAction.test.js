import configureStore from "redux-mock-store";
import * as UserActions from "../../src/actions/userAction";
import MockAdapter from "axios-mock-adapter";
import thunk from "redux-thunk";
import axios from "axios";

const mockStore = configureStore([thunk]);
const initState = {
  userReducer: {
    user: {
      username: "",
      sex: "男",
      age: 20
    },
    list: []
  },
  actions: {
    changeUsername: UserActions.changeUsername,
    changeUserSex: UserActions.changeUserSex
  }
};
const store = mockStore(initState);
const axiosMock = new MockAdapter(axios);

describe("user actions", () => {
  beforeEach(() => {
    store.clearActions();
  });

  test("dispatch correct actoin and payload", () => {
    const expectedActions = [
      {
        type: "CHANGE_USERNAME",
        payload: "abc"
      },
      {
        type: "CHANGE_SEX",
        payload: true
      }
    ];
    store.dispatch(UserActions.changeUsername("abc"));
    store.dispatch(UserActions.changeUserSex(true));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  test("test async action", async () => {
    const mockData = {
      status: 200,
      data: {
        code: 0,
        data: [
          {
            username: "tom",
            sex: "男",
            age: 18
          },
          {
            username: "jack",
            sex: "男",
            age: 30
          },
          {
            username: "lucy",
            sex: "女",
            age: 25
          }
        ],
        msg: ""
      }
    };
    axiosMock
      .onGet("http://www.mocky.io/v2/5c7658db3200000b24f46080")
      .reply(200, mockData);
    const expectAction = {
      type: "GET_ALL_USERS",
      payload: mockData.data.data
    };
    store.dispatch(UserActions.getAllUsers()).then(() => {
      expect(store.getActions()).toContainEqual(expectAction);
    });
    await store.dispatch(UserActions.getAllUsers());
    expect(store.getActions()).toContainEqual(expectAction);
  });
});
