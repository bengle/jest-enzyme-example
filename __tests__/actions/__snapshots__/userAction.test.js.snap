// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user actions dispatch correct actoin and payload 1`] = `
Array [
  Object {
    "payload": "abc",
    "type": "CHANGE_USERNAME",
  },
  Object {
    "payload": true,
    "type": "CHANGE_SEX",
  },
]
`;
